qcengine/programs/entos.py

```diff
@@ -0,0 +1,238 @@
+            input_file = "\n".join(input_file)
+        else:
+            import jinja2
+            # Some of the potential different template options
+            # (A) ordinary build_input (need to define a base template)
+            # (B) user wants to add stuff after normal template (A)
+            # (C) user knows their domain language (doesn't use any QCSchema quantities)
+
+            # # Build dictionary for substitute
+            # sub_dict = {
+            #     "method": input_model.model.method,
+            #     "basis": input_model.model.basis,
+            #     "df_basis": input_model.keywords["df_basis"].upper(),
+            #     "charge": input_model.molecule.molecular_charge
+            # }
+
+            # Perform substitution to create input file
+            str_template = jinja2.Template(template)
+            input_file = str_template.render()
+
+        return {
+            "commands": ["entos", "-n", str(config.ncores), "dispatch.in"],
+            "infiles": {
+                "dispatch.in": input_file,
+                "geometry.xyz": xyz_file
+            },
+            "scratch_location": config.scratch_directory,
+            "input_result": input_model.copy(deep=True)
+        }
+
+    def write_input_recursive(self, d: Dict[str, Any]) -> List:
+        input_file = []
+        for key, value in d.items():
+            if isinstance(value, dict):
+                input_file.append(key + '(')
+                rec_input = self.write_input_recursive(value)
+                indented_line = map(lambda x: "  " + x, rec_input)
+                input_file.extend(indented_line)
+                input_file.append(')')
+            else:
+                if isinstance(value, str):
+                    input_file.append("{0} = '{1}'".format(key, value))
+                elif isinstance(value, bool):
+                    input_file.append("{0} = {1}".format(key, str(value).lower()))
+                else:
+                    input_file.append("{0} = {1}".format(key, value))
+        return input_file
+
+    def parse_output(self, outfiles: Dict[str, str], input_model: 'ResultInput') -> 'Result':
+
+        output_data = {}
+        properties = {}
+
+        # Parse the output file, collect properties and gradient
+        output_lines = outfiles["dispatch.out"].split('\n')
+        gradients = []
+        natom = len(input_model.molecule.symbols)
+        for idx, line in enumerate(output_lines):
+            fields = line.split()
+            if fields[:1] == ["energy:"]:
+                properties["scf_total_energy"] = float(fields[-1])
+            elif fields[:2] == ["Molecular", "Dipole:"]:
+                properties["scf_dipole_moment"] = [float(x) for x in fields[2:5]]
+            elif fields[:3] == ["SCF", "converged", "in"]:
+                properties["scf_iterations"] = int(fields[3])
+            elif fields == ["Gradient", "(hartree/bohr):"]:
```

Comment:
Since Entos is new, I don't suppose we could convince them to drop JSON output or even CSV?

Reply:
They are in the process of writing their output in XML and/or JSON. It isn't currently done, but once it is I'll update the output parser.